src/backend/distributed/commands/alter_table.c

```diff
@@ -441,7 +445,11 @@ AlterDistributedTable(TableConversionParameters *params)
 		ereport(DEBUG1, (errmsg("setting multi shard modify mode to sequential")));
 		SetLocalMultiShardModifyModeToSequential();
 	}
-	return ConvertTable(con);
+	SetupReplicationOriginLocalSession();
```

Comment:
alter_distributed_table is converting from one distributed table to another, so the writes happen remotely, not locally.

Reply:
This might be fairly complex, since the write is done through an INSERT..SELECT command that runs via the executor. I'm not quite sure how best to approach this in an unintrusive fashion.

Reply:
With the current changes, if the shard is created in the co-ordinator node, those changes are supporesed. But any changes in remote nodes are not skipped. If I know where is the remote session created for this in code, I can add SetupRemoteReplicationOrigin() to skip the changes.. It is not going through the ShardCopyDestReceiver code..

Reply:
maybe we omit this altogether and document that alter_distributed_table re-emits the table as a long list of inserts. It would be a bit confusing if it behaves differently depending on where the shards are.